### DISCUSSION TITLE
Subfield Naming in Topic Schema for 'record' Data Types

### DISCUSSION THREAD

**Discussion by andyecoh** [cloudera/hue#3202] [Questions]
Hi Hue team – my team uses Hue to run FlinkSQL queries on our Pulsar topics, and some of the protobuf messages on our topics have instances of `record` data types. For example, we have topics that look like this when I drill into them in the left panel in Hue: 

my_topic:
- my_record_field (row)
- my_bigint_field (bigint)
- my_string_field (string)

I am wondering how complicated it would be to be able to view _subfields_ from record types in this same view. To be able to see something like: 

my_topic:
- **my_record_field.my_subfield_1 (double)**
- **my_record_field.my_subfield_2 (int)**
- my_bigint_field (bigint)
- my_string_field (string)

Has this been discussed previously? Or have others implemented workarounds for sub-messages in tables/topics? Thanks!

**Comment by andyecoh** [comment]
This is still relevant. @romainr I see you're a community contributor – do you know any codeowners who would be able to help me out with this question? Thanks!

**Comment by bjornalm** [comment]
Hi @andyecoh, thanks for engaging and sorry for the very late reply. We are working on getting our community process back on track. I'm gonna start by turning this issue into a question in our discussions forum and see if anyone there can help out.
